egov-pg-service/src/main/java/org/egov/pg/service/gateways/axis/AxisGateway.java

```diff
@@ -85,6 +85,9 @@ public AxisGateway(RestTemplate restTemplate, Environment environment, ObjectMap
     @Override
     public URI generateRedirectURI(Transaction transaction) {
 
+        Map metaData = transaction.getMetaData();
```

Comment:
This looks wrong somehow. The `metaData` is related to Gateway and not a transaction as such. During a transaction you want meta data related to the gateway?

Reply:
Sort of, i will take the gateway and other details and put the req metadata in the transaction itself so it can be used. i think nitin told me to do it this way.